### PR TITLE
[otbn] Handle memory errors in the ISS and re-enable smoke test

### PR DESCRIPTION
**Pull request by rswarbrick**

### CODE REVIEW

**Comment by rswarbrick** [comment]
Force-push adds support for getting the `err_code` value from the internals of the model, out to the wider RTL simulation. This is needed to get things right when running a chip simulation with a model and no OTBN RTL when it runs a program that generates an error code. Of course, I've also wired things up properly in the Verilator block-level testbench so that it checks that the error code gets set properly.

**Comment by rswarbrick** [comment]
Note that the trace handling changes in this PR clash with those in #4614. Whichever one gets merged second will need a bit of rejigging in the ISS wrapper.

File: hw/ip/otbn/dv/model/iss_wrapper.cc [hunk 1]
```diff
@@ -189,6 +189,37 @@ static uint32_t read_hex_32(const char *str) {
...
+  //
+  //   otbn.$REG_NAME &= ~ 0x000001 (from HW) (now 0x000000)
+  //
+  // The \n picks up the newline that we expect at the end of each line.
+  std::regex re("\\s*otbn\\." + reg_name + ".*0x([0-9a-f]{8})\\)\n");
+  std::smatch match;
+
+  uint32_t val = default_val;
+
+  std::cerr << "read_ext_reg(" << reg_name << ", ...)\n";
```
Comment (imphil) [inline hw/ip/otbn/dv/model/iss_wrapper.cc:206]: Is this leftover debug output (also the other std::cerr lines in this function)?

File: hw/ip/otbn/dv/model/iss_wrapper.cc [hunk 2]
```diff
@@ -189,6 +189,37 @@ static uint32_t read_hex_32(const char *str) {
   return strtoul(buf, nullptr, 16);
 }
 
+static uint32_t read_ext_reg(const std::string &reg_name,
```
Comment (imphil) [inline hw/ip/otbn/dv/model/iss_wrapper.cc:192]: Maybe add a short comment that this function reads a bus-accessible CSR?

**Review by imphil — APPROVED** [review]

File: hw/ip/otbn/dv/model/iss_wrapper.cc (hunk 1 above)
Comment (rswarbrick) [reply hw/ip/otbn/dv/model/iss_wrapper.cc:206]: Oops! Thanks!

**Pull request merged**